### PR TITLE
chore(demo-app): code refactor & enhancements

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Enhance code refactor for the Demo App for better readability and  maintainability

## How did you test it?

Making successful payments and checking for basic edge cases

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by aritro2002 — APPROVED** [review]

File: Hyperswitch-React-Demo-App/src/Payment.js [hunk 1]
```diff
@@ -1,105 +1,123 @@
...
-      ]);
+      const configData = await configRes.json();
+      const urlsData = await urlsRes.json();
+
+      const paymentIntentData = isCypressTestMode
+        ? { clientSecret: clientSecretQueryParam }
+        : await fetch(`${baseUrl}/create-payment-intent`).then((res) => {
+            if (!res.ok) throw new Error("Failed to fetch payment intent");
+            return res.json();
+          });
```
[the remark this answers is not in the document]
Comment (PritishBudhiraja) [reply Hyperswitch-React-Demo-App/src/Payment.js]: Updated

**Review by Sanskar2001 — DISMISSED** [review]

File: Hyperswitch-React-Demo-App/src/CheckoutForm.js [hunk 2]
```diff
@@ -1,121 +1,87 @@
...
 
   const handleSubmit = async (e) => {
     e.preventDefault();
 
-    if (!hyper || !elements) {
-      // Hyper.js has not yet loaded.
-      // Make sure to disable form submission until Hyper.js has loaded.
-      return;
-    }
+    if (!hyper || !elements || isProcessing) return;
```
[the remark this answers is not in the document]
Comment (PritishBudhiraja) [reply Hyperswitch-React-Demo-App/src/CheckoutForm.js:33]: Updated

**Review by Sanskar2001 — APPROVED** [review]

**Review by aritro2002 — APPROVED** [review]
